src/components/HotRepoCard.tsx

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useState } from "react";
+import { RiCheckboxCircleFill } from "react-icons/ri";
+import { FaArrowAltCircleUp } from "react-icons/fa";
+import { AiOutlineStar } from "react-icons/ai";
+import { BiGitPullRequest } from "react-icons/bi";
+import { VscIssues } from "react-icons/vsc";
+import Skeleton from "react-loading-skeleton";
+import { getAvatarLink } from "../lib/github";
+import humanizeNumber from "../lib/humanizeNumber";
+import StackedAvatar from "./StackedAvatar";
+import useRepo from "../hooks/useRepo";
+import useVotedRepos from "../hooks/useVotedRepos";
+
+export declare interface HotRepoCardProps {
+  repoName: string;
+}
+
+const HotRepoCard = ({ repoName }: HotRepoCardProps): JSX.Element => {
+  const { votedReposIds, checkVoted, voteHandler } = useVotedRepos();
+  const { repo, isLoading, isError } = useRepo(repoName);
+  const [isVoted, setIsVoted] = useState(false);
+
+  useEffect(() => {
+    repo && setIsVoted(checkVoted(repo.id));
+  }, [votedReposIds, repo]);
+
+  if (isError) {
+    return (
+      <div className="p-4 border rounded-2xl bg-white w-full space-y-1 relative">
+        {`${repoName} failed to load`}
+      </div>
+    );
+  }
+
+  if (isLoading) {
+    return (
+      <div className="p-4 border rounded-2xl bg-white w-full space-y-1 relative">
+        <Skeleton
+          enableAnimation
+          count={5}
+        />
+      </div>
+    );
+  }
+
+  const { id, full_name, name, description, issues, stars, contributions } = repo!;
```

Comment:
What happens when if one of these properties is null? I think You have to treat this.